src/XPurchase/VirtoCommerce.XPurchase/Schemas/GiftItemType.cs

```diff
@@ -0,0 +1,56 @@
+using System.Linq;
+using GraphQL.DataLoader;
+using GraphQL.Resolvers;
+using GraphQL.Types;
+using MediatR;
+using VirtoCommerce.CartModule.Core.Model;
+using VirtoCommerce.ExperienceApiModule.Core.Extensions;
+using VirtoCommerce.ExperienceApiModule.Core.Helpers;
+using VirtoCommerce.ExperienceApiModule.Core.Schemas;
+using VirtoCommerce.ExperienceApiModule.Core.Services;
+using VirtoCommerce.XDigitalCatalog;
+using VirtoCommerce.XDigitalCatalog.Queries;
+using VirtoCommerce.XDigitalCatalog.Schemas;
+
+namespace VirtoCommerce.XPurchase.Schemas
+{
+    public class GiftItemType : ExtendableGraphType<LineItem>
+    {
+        public GiftItemType(IMediator mediator, IDataLoaderContextAccessor dataLoader, IDynamicPropertyResolverService dynamicPropertyResolverService)
+        {
+            var productField = new FieldType
+            {
+                Name = "product",
+                Type = GraphTypeExtenstionHelper.GetActualType<ProductType>(),
+                Resolver = new FuncFieldResolver<LineItem, IDataLoaderResult<ExpProduct>>(context =>
+                {
+                    var includeFields = context.SubFields.Values.GetAllNodesPaths();
+                    var loader = dataLoader.Context.GetOrAddBatchLoader<string, ExpProduct>("order_lineItems_products", async (ids) =>
+                    {
+                        //Gift is not part of cart, can't use CartAggregate. Getting store and currency from the context.
+                        var request = new LoadProductsQuery
+                        {
+                            StoreId = context.GetValue<string>("storeId"),
+                            CurrencyCode = context.GetArgumentOrValue<string>("currencyCode"),
+                            ObjectIds = ids.ToArray(),
+                            IncludeFields = includeFields.ToArray()
+                        };
+
+                        var response = await mediator.Send(request);
+
+                        return response.Products.ToDictionary(x => x.Id);
+                    });
+                    return loader.LoadAsync(context.Source.ProductId);
```

Comment:
Missing await?

Reply:
no, the expected IDataLoaderResult<> is returned